discovery-provider/src/tasks/index_plays.py

```diff
@@ -0,0 +1,123 @@
+                        func.count(Play.play_item_id)
+                    ).filter(
+                        Play.play_item_id == listen['trackId'],
+                        Play.user_id == listen['userId']
+                    ).scalar()
+                    new_play_count = listen['count'] - user_track_play_count
+                    if new_play_count > 0:
+                        plays.extend([
+                            Play(
+                                user_id=listen['userId'],
+                                play_item_id=listen['trackId'],
+                                created_at=listen['createdAt'],
+                            ) for _ in range(new_play_count)
+                        ])
+                else:
+                    # For anon track plays, check the current hour play counts
+                    # and only insert new plays for the difference
+                    current_hour_query = dateutil.parser.parse(
+                            listen['createdAt']
+                        ).replace(microsecond=0, second=0, minute=0)
+                    anon_hr_track_play_count = session.query(
+                        func.count(Play.play_item_id)
+                    ).filter(
+                        Play.play_item_id == listen['trackId'],
+                        Play.user_id == None,
+                        Play.created_at >= current_hour_query
+                    ).scalar()
+                    new_play_count = listen['count'] - anon_hr_track_play_count
+                    if new_play_count > 0:
+                        plays.extend([
+                            Play(
+                                play_item_id=listen['trackId'],
+                                created_at=listen['createdAt'],
+                            ) for _ in range(new_play_count)
+                        ])
+
+        if len(plays) > 0:
+            session.bulk_save_objects(plays)
+            session.execute("REFRESH MATERIALIZED VIEW aggregate_plays")
+
+######## CELERY TASK ########
+@celery.task(name="update_play_count", bind=True)
+def update_play_count(self):
+    # Cache custom task class properties
+    # Details regarding custom task context can be found in wiki
+    # Custom Task definition can be found in src/__init__.py
+    db = update_play_count.db
+    redis = update_play_count.redis
+
+    # Define lock acquired boolean
+    have_lock = False
+
+    # Define redis lock object
+    update_lock = redis.lock("update_play_count_lock", blocking_timeout=25)
+    try:
+        # Attempt to acquire lock - do not block if unable to acquire
+        have_lock = update_lock.acquire(blocking=False)
+        if have_lock:
+            logger.info(f"index_plays.py | update_play_count | {self.request.id} | Acquired update_play_count_lock")
+            get_track_plays(self, db)
+            logger.info(f"index_plaus.py | update_play_count | {self.request.id} | Processing complete within session")
```

Comment:
minor nit - let's make these plays instead of index_plaus

Reply:
plaussssssssssssssssssssssssss